test/cli/daemon.js

```diff
@@ -81,31 +84,42 @@ describe('daemon', () => {
         Gateway: '/ip4/127.0.0.1/tcp/0'
       }), '--json')
     }).then(() => {
-      return ipfs('daemon')
-    }).then((res) => {
-      expect(res).to.have.string('Daemon is ready')
-      done()
+      const proc = ipfs('daemon')
+      pull(
+        toPull(proc.stdout),
+        pull.collect((err, res) => {
+          expect(err).to.not.exist()
+          const data = res.toString()
+          expect(data).includes('Daemon is ready')
+          proc.kill()
+          done()
+        })
+      )
```

Comment:
This needs to be reviewed. Looks like testing the tests code and not really necessary AFAIK.